code/mathematical_algorithms/src/average_stream_numbers/average_stream_numbers.rb

```diff
@@ -0,0 +1,26 @@
+# Part of Cosmos by OpenGenus Foundation
+
+def average_stream_number(input_array)
+  avg = 0
+  resultant_array = []
+  input_array.each_with_index do |element, i|
+    avg = average(element, avg, (i + 1))
+    puts avg
+    resultant_array << avg
+  end
+  resultant_array
```

Comment:
What purpose does the return serve? Also, instead of printing out the average, why not yield to a block each new computed average instead?